### DISCUSSION TITLE
Can zip.js run in Node.js?

### DISCUSSION THREAD

**Discussion by ArndBrugman** [gildas-lormeau/zip.js#288] [Q&A] [answered]
Hi, This looks like a great solution for one of my projects, performance looks to be impressive. Is this code compatible to be shared with both front end webdevelopment using React and Typescript and back end development (service) using NodeJS and TypeScript?

**Comment by gildas-lormeau** [comment]
Unfortunately, zip.js was not designed to run in Node.js. However, it might work by adding some polyfills. For example, https://www.npmjs.com/package/web-worker would be needed to emulate web workers. Note that zip.js works fully in Deno though. I'll try to do some tests to find the polyfills needed to run zip.js in Node.js.

**Reply by gildas-lormeau** [reply]
BTW, you might also be interested by fflate: https://github.com/101arrowz/fflate

**Comment by gildas-lormeau** [comment, marked as answer]
I confirm it's possible to use zip.js (v2.3.19+) in Node.js (v14.8+) by polyfilling the `Blob` and the `Worker` APIs, see this example https://github.com/gildas-lormeau/zip.js/blob/3b9d2f2d789c95cddb4787f42a120c11b88dc7c5/tests/node/. The code in the `test()` function should run on client-side (i.e. in the browser) or server-side (in Node.js) without needing any change.

**Reply by ArndBrugman** [reply]
Thx for taking the time to add this. Due to my current development platform targets I am sadly limited to node ~10 max right now. I was unsuccessful in a quick attempt to get your example running. I'll try your previous suggestion and take a look at https://github.com/101arrowz/fflate. Thx!

**Reply by gildas-lormeau** [reply]
no problem

**Reply by ArndBrugman** [reply]
I was able to implement both server (NodeJS) and client (React) with your mentioned alternative. Thx for your time and have a good weekend!

**Reply by gildas-lormeau** [reply]
Thank you for the feedback, have a good weekend too!
